src/python/foglamp/core/server.py

```diff
@@ -40,15 +55,118 @@ def _make_app():
         routes.setup(app)
         return app
 
+    @staticmethod
+    def _make_core_app():
+        """Creates the Service management REST server Core a.k.a. service registry
+
+        :rtype: web.Application
+        """
+        app = web.Application(middlewares=[middleware.error_middleware])
+        routes.core_setup(app)
+        return app
+
     @classmethod
     async def _start_scheduler(cls):
         """Starts the scheduler"""
-        cls.scheduler = Scheduler()
+        _logger.info("start scheduler")
+        cls.scheduler = Scheduler(cls.core_management_port)
         await cls.scheduler.start()
 
+    @staticmethod
+    def __start_storage(host, m_port):
+        _logger.info("start storage, from directory %s", _STORAGE_DIR)
+        try:
+            cmd_with_args = ['./storage', '--address={}'.format(host),
+                             '--port={}'.format(m_port)]
+            subprocess.call(cmd_with_args, cwd=_STORAGE_DIR)
+        except Exception as ex:
+            _logger.exception(str(ex))
+
+    @classmethod
+    async def _start_storage(cls, loop):
```

Comment:
Correct me if I am wrong, __start_storage and _start_storage can be collapsed into a single method. No need to to run call_soon anymore as we are running run_until_complete on the parent call:

loop.run_until_complete(cls._start_storage(loop)) can now be loop.run_until_complete(cls._start_storage())

This would simplify the code.

Reply:
Will try to figure out `Async and await with subprocess with cd to specific directoy`. Here we are using subprocess.call which allows cwd for command to be executed in that subprocess which  is `__start_storage` callback method here. Hence kept `_start_storage` as async wrapper here to call in the loop. Please feel free to commit on top of this, if we have got some better solution already.